docs/source/azure/configuration/serverlessfunctions.rst

```diff
@@ -18,6 +18,28 @@ Python support in Azure Functions V2 is in preview and this feature is still imm
 Currently periodic (CRON) and Event Grid functions are supported, however consumption pricing is not
 yet supported.
 
+Azure Modes
```

Comment:
So the modes aren't necessarily all azure functions related, but admittedly all the ones that matter are.

pull, the default, is just running at your local CLI.  

We may be adding some more modes around Azure Graph support soon though, or other non-functions hosting options.

I think this can all go in this page, but we might think about making it a more generic `run modes` or `hosting` type thing that covers all the ways you can run (serverless, local, future options) - instead of making it all about functions.